Buttplug.Core/Messages/Messages.cs

```diff
@@ -411,6 +523,157 @@ public SingleMotorVibrateCmd(uint aDeviceIndex, double aSpeed, uint aId = Buttpl
         }
     }
 
+    public class VibrateCmd : ButtplugDeviceMessage
+    {
+        public class VibrateIndex
+        {
+            private double _speedImpl = 0;
+
+            [JsonProperty(Required = Required.Always)]
+            public uint Index = 0;
+
+            [JsonProperty(Required = Required.Always)]
+            public double Speed
+            {
+                get => _speedImpl;
+                set
+                {
+                    if (value < 0)
+                    {
+                        throw new ArgumentException("VibrateCmd Speed cannot be less than 0!");
+                    }
+
+                    if (value > 1)
+                    {
+                        throw new ArgumentException("VibrateCmd Speed cannot be greater than 1!");
+                    }
+
+                    _speedImpl = value;
+                }
+            }
+
+            public VibrateIndex(uint aIndex, double aSpeed)
+            {
+                Index = aIndex;
+                Speed = aSpeed;
+            }
+        }
+
+        [JsonProperty(Required = Required.Always)]
+        public List<VibrateIndex> Speeds;
+
+        public VibrateCmd(uint aDeviceIndex, List<VibrateIndex> aSpeeds, uint aId = ButtplugConsts.DefaultMsgId)
+            : base(aId, aDeviceIndex)
+        {
+            Speeds = aSpeeds;
+            MessageVersioningVersion = 1;

```

Comment:
I originally implemented this as static members, but the inheritance didn't seem to be working for me: because I was dealing with the base ButtplugMessage, it really didn't want to give me the subclass value, just the default base class.